### DISCUSSION TITLE
fetching data client-side

### DISCUSSION THREAD

**Discussion by acidjazz** [nuxt/nuxt#18137] [Questions] [answered]
Any recommendations for fetching data client-side with a loading state?

I tried using `$fetch` in `onMounted(()` but got the warning:

```[Vue warn]: onMounted is called when there is no active component instance to be associated with. Lifecycle injection APIs can only be used during execution of setup(). If you are using async setup(), make sure to register lifecycle hooks before the first await statement.```

I used `<script lang="ts" setup>` .. arent we inside setup then?

**Comment by ivanq3w** [comment]
You can try to fetch with server: false option, https://v3.nuxtjs.org/docs/usage/data-fetching#useasyncdata
```
<script setup lang="ts">
const { data: items } = await useFetch(`https://myapi.com/items`, { server: false })
</script>
```
And you cant use composable in onMounted() cos it need to be in setup function

**Comment by acidjazz** [comment, marked as answer]
@ivanq3w I think I see what you mean.. so something lke:

```ts

const {
  data: result,
  refresh: reloadExample,
}: {
  data: Ref<api.MetApiResults & { data: Examples }>
  refresh: (force?: boolean) => Promise<void>
} = useAsyncData(
  'example',
  () => $api.index<Examples>('/example', { count: 3 }),
  { server: false },
)

```

**Reply by ivanq3w** [reply]
Yep
